android/app/src/main/java/app/organicmaps/settings/SettingsPrefsFragment.java

```diff
@@ -144,6 +147,17 @@ private void enableListeners(boolean enable)
       mTtsPrefEnabled.setOnPreferenceChangeListener(enable ? mEnabledListener : null);
     if (mTtsPrefLanguages != null)
       mTtsPrefLanguages.setOnPreferenceChangeListener(enable ? mLangListener : null);
+    if (mTtsPrefStreetNames != null) {
+      mTtsPrefStreetNames.setOnPreferenceChangeListener(new Preference.OnPreferenceChangeListener() {
+        @Override
+        public boolean onPreferenceChange(Preference preference, Object newValue) {
+          boolean newVal = (Boolean) newValue;
```

Comment:
Can this value change to a non-boolean type in the future? Is it worth checking the type conversion dynamically to avoid crashes in the future after a type change?

Reply:
I can't imagine a situation where we'd want to make "announce street names" much besides yes or no. And in any case if we did we'd need to go in and reimplement `Config.setAnnounceStreets(newVal)` as well as `mTtsPrefStreetNames` and test that the change worked... seems fine as is to me but I'm not a professional Java programmer...